packages/tooltip/src/Tooltip.spec.tsx

```diff
@@ -53,6 +65,67 @@ describe('packages/tooltip', () => {
       expect(tooltip).not.toBeVisible();
     });
 
+    test('when "triggerEvent" is set to "hover", hovering on and off the trigger opens and closes the tooltip', async () => {
+      const { getByTestId, queryByTestId, button } = renderTooltip({
+        triggerEvent: 'hover',
+      });
+
+      fireEvent.mouseEnter(button);
+
+      await waitFor(() => getByTestId(tooltipTestId), { timeout: 500 });
+
+      expect(getByTestId(tooltipTestId)).toBeInTheDocument();
+
+      fireEvent.mouseLeave(button);
+
+      await waitForElementToBeRemoved(getByTestId(tooltipTestId), {
+        timeout: 500,
+      });
+
+      expect(queryByTestId(tooltipTestId)).not.toBeInTheDocument();
+    });
+
+    const fireEventMap = {
+      hover: {
+        on: fireEvent.mouseEnter,
+        off: fireEvent.mouseLeave,
+      },
+      click: {
+        on: fireEvent.click,
+        off: fireEvent.click,
+      },
+    };
+
+    async function testTriggerEventWhenDisabled(
+      triggerEvent: 'hover' | 'click',
+    ) {
+      test(`when triggerEvent is "${triggerEvent}"`, async () => {
+        const { queryByTestId, button } = renderTooltip({
+          triggerEvent,
+          enabled: false,
+        });
+
+        fireEventMap[triggerEvent].on(button);
+
+        // Wait for 200ms to allow the component to animate out and remove the element from the DOM
+        await waitForTimeout(200);
```

Comment:
Can we use `waitFor` here instead, and wait until the element is removed from the DOM? This seems more brittle than declaring what we're waiting for.

Reply:
This test is specifically testing that nothing changes when the trigger is interacted with, so there's no DOM change to wait for in a passing case.